src/lang/java.rs

```diff
@@ -0,0 +1,579 @@
+                    &mut methods,
+                    &mut fields,
+                );
+                start = _start as i32;
+                end = _end as i32;
+            }
+            unknown_type => {
+                eprintln!(
+                    "{} tag unhandled. This may or may not be an issue.",
+                    unknown_type
+                );
+            }
+        };
+    }
+
+    // Assemble the return value from what was extracted from the tree
+    Some(ClassOrInterfaceComponent {
+        component: ContainerComponent {
+            component,
+            accessor: modifier.accessor,
+            stereotype: stereotype,
+            methods,
+            container_name: instance_name,
+            line_count: end - start,
+        },
+        declaration_type: ContainerType::Class,
+        annotations: modifier.annotations,
+        constructors: fold_vec(constructors),
+        field_components: fold_vec(fields),
+    })
+}
+
+/// Parse the AST for a specified method
+fn parse_method(ast: &AST, component: &ComponentInfo) -> MethodComponent {
+    // Define fields
+    let mut body = None;
+    let mut modifier = Modifier::new();
+    let mut method_name = String::new();
+    let mut parameters = vec![];
+    let return_type = find_type(ast);
+    let line_begin = ast.span.expect("No span for a method! AST malformed!").0 as i32;
+    let line_end = ast.span.expect("No span for a method! AST malformed!").2 as i32;
+
+    // Parse method
+    for member in ast.children.iter() {
+        match &*member.r#type {
+            "identifier" | "static" => method_name = member.value.clone(),
+            "modifiers" => modifier = parse_modifiers(member),
+            "formal_parameters" => parameters = parse_method_parameters(member, component),
+            "constructor_body" | "block" => {
+                body = Some(parse_block(member, component));
+            }
+            unknown => println!("{} unknown", unknown),
+        }
+    }
+
+    // Return the method component
+    MethodComponent {
+        component: component.clone(),
+        accessor: modifier.accessor,
+        method_name,
+        return_type,
+        parameters,
+        is_static: modifier.is_static,
+        is_abstract: modifier.is_abstract,
+        is_final: modifier.is_final,
+        sub_methods: vec![],
+        annotations: modifier.annotations,
+        line_count: line_end - line_begin,
+        line_begin,
+        line_end,
+        body,
+    }
+}
+
+/// Struct to hold return data from parse_modifiers
+struct Modifier {
+    accessor: AccessorType,
+    annotations: Vec<AnnotationComponent>,
+    is_abstract: bool,
+    is_final: bool,
+    is_static: bool,
+}
+impl Modifier {
+    fn new() -> Modifier {
+        Modifier {
+            accessor: AccessorType::Default,
+            annotations: vec![],
+            is_abstract: false,
+            is_final: false,
+            is_static: false,
+        }
+    }
+}
```

Comment:
Looks like what you're trying to do here is to create a default Modifier, this is actually what the `Default` trait is for. You can `derive(Default)` to generate the `Modifier::default()` function for you.

Not a big deal, just figured I would point it out.

Reply:
Do note that to derive Default everything else needs to implement Default as well, and since AccessorType is an enum it would need a manual implementation